### PR TITLE
Check cluster(s) before delete a secret

### PR DESCRIPTION
**Pull request by colin014**
#314

### CODE REVIEW

File: api/cluster.go [hunk 1]
```diff
@@ -58,7 +58,18 @@ func GetCommonClusterFromRequest(c *gin.Context) (cluster.CommonCluster, bool) {
 		})
 		return nil, false
 	}
-	commonCLuster, err := cluster.GetCommonClusterFromModel(modelCluster)
+
+	if len(modelCluster) == 0 {
+		log.Errorf("Empty cluster list")
```
Comment (baluchicken) [inline api/cluster.go]: log.Error

File: api/secrets.go [hunk 2]
```diff
@@ -184,3 +195,25 @@ func IsValidSecretType(secretType string) error {
...
+// checkClustersBeforeDelete returns error if there's a running cluster that created with the given secret
+func checkClustersBeforeDelete(orgId, secretId string) error {
+
+	filter := map[string]interface{}{
+		"organization_id": orgId,
+		"secret_id":       secretId,
+	}
+
+	if modelCluster, err := model.QueryCluster(filter); err != nil {
+		return nil
```
Comment (baluchicken) [inline api/secrets.go]: Handle error properly here.

File: api/secrets.go [hunk 3]
```diff
@@ -184,3 +195,25 @@ func IsValidSecretType(secretType string) error {
...
+	} else {
+		for _, mc := range modelCluster {
+			if commonCluster, err := cluster.GetCommonClusterFromModel(&mc); err == nil {
+				if _, err := commonCluster.GetStatus(); err == nil {
+					return errors.New(fmt.Sprintf("There's a running cluster with this secret: %s[%d]", mc.Name, mc.ID))
```
Comment (baluchicken) [inline api/secrets.go]: Use fmt.Errorf instead of errors.new and fmt.Sprintf

File: secret/store.go [hunk 4]
```diff
@@ -157,6 +157,11 @@ func (ss *secretStore) Get(organizationID string, secretID string) (*SecretsItem
 	if err != nil {
 		return nil, err
 	}
+
+	if secret == nil {
+		return nil, errors.New(fmt.Sprintf("There's no secret with this id: %s", secretID))
```
Comment (baluchicken) [inline secret/store.go]: Use fmt.Errors

File: api/secrets.go (hunk 2 above)
Comment (colin014) [reply api/secrets.go]: When an error occours here that means no cluster found the database with the given params and the secret can be deleted so the method sould be return nil error here. But i add an info log before return

**Pull request merged**